ccl/crosscluster/logical: TestFourWayReplication failed

ccl/crosscluster/logical.TestFourWayReplication [failed](https://mesolite.cluster.engflow.com/invocations/default/dfb3c626-5d9d-4991-bdf3-0cd82b3dc36a?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [028f96fbccf525df470e41ef58797935393e4c5d](https://github.com/cockroachdb/cockroach/commits/028f96fbccf525df470e41ef58797935393e4c5d):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFourWayReplication4205326884
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1020815129178767361 to reach replicated time of 1731598457.448267844,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020815129825280001 to reach replicated time of 1731598457.448267844,0
    jobs_verification.go:218: error scanning '&{<nil> 0xc00483b7a0}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestFourWayReplication4205326884
--- FAIL: TestFourWayReplication (4.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44393

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17765458?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17765458?buildTab=artifacts#/) on master @ [601cf87ad5e672ed785e7d6ea6ac06c0c3ea06fe](https://github.com/cockroachdb/cockroach/commits/601cf87ad5e672ed785e7d6ea6ac06c0c3ea06fe):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication144159670
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863641985974273 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863642229407745 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863642421886977 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863642597687297 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863642716569601 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863642864844801 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863642978942977 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863643264155649 to reach replicated time of 1731613260.087007316,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1020863643415019521 to reach replicated time of 1731613260.087007316,0
    jobs_verification.go:218: error scanning '&{<nil> 0x40090b1cb0}': sql: no rows in result set
    panic.go:626: -- test log scope end --
--- FAIL: TestFourWayReplication (1.60s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17777865?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17777865?buildTab=artifacts#/) on master @ [ad45b52e0ec40cd2ba8e79510e1d0d6aebcb2041](https://github.com/cockroachdb/cockroach/commits/ad45b52e0ec40cd2ba8e79510e1d0d6aebcb2041):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication2735900572
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079632892067841 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633003905025 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633116954625 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633228562433 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633340301313 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633447911425 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633548214273 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079633644847105 to reach replicated time of 1731679175.187893546,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021079634165792769 to reach replicated time of 1731679175.187893546,0
    jobs_verification.go:218: error scanning '&{<nil> 0x4007035680}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication2735900572
--- FAIL: TestFourWayReplication (1.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17782215?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17782215?buildTab=artifacts#/) on master @ [d06e762c101cb29015c7db49f9b2d819e2483df1](https://github.com/cockroachdb/cockroach/commits/d06e762c101cb29015c7db49f9b2d819e2483df1):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication3288143934
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111672381702145 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111672513822721 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111672659738625 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111672833638401 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111672991318017 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111673099255809 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111673206702081 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111673317720065 to reach replicated time of 1731688952.984779767,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021111673920126977 to reach replicated time of 1731688952.984779767,0
    jobs_verification.go:218: error scanning '&{<nil> 0x4006a047e0}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication3288143934
--- FAIL: TestFourWayReplication (1.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3 release-blocker]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://tanzanite.cluster.engflow.com/invocations/default/6462158b-439f-4f6e-82be-c736dbd58371?testReportRun=16&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL2Nyb3NzY2x1c3Rlci9sb2dpY2FsOmxvZ2ljYWxfdGVzdA==) on master @ [e83bc46aa42f2476b4b11b9703b8038c660dc980](https://github.com/cockroachdb/cockroach/commits/e83bc46aa42f2476b4b11b9703b8038c660dc980):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestFourWayReplication1323602432
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363254107963393 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363254346711041 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363255376543745 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363255966203905 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363256451104769 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363256949440513 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363257758810113 to reach replicated time of 1731765731.514688560,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021363258357252097 to reach replicated time of 1731765731.514688560,0
    jobs_verification.go:218: error scanning '&{<nil> 0xc00c84f0e0}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestFourWayReplication1323602432
--- FAIL: TestFourWayReplication (3.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=16</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3 release-blocker]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17819322?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17819322?buildTab=artifacts#/) on master @ [5b882075878be4a8e8106c6af33dce96daf4e3a2](https://github.com/cockroachdb/cockroach/commits/5b882075878be4a8e8106c6af33dce96daf4e3a2):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication462062243
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956689267523585 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956689849384961 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956690174803969 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956690599608321 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956690769510401 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956691121012737 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956691728957441 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956691961184257 to reach replicated time of 1731946832.427223614,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1021956692322975745 to reach replicated time of 1731946832.427223614,0
    jobs_verification.go:218: error scanning '&{<nil> 0x400bb3c000}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication462062243
--- FAIL: TestFourWayReplication (3.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17862596?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17862596?buildTab=artifacts#/) on master @ [b3f25323eb94ac2759f2b51f440050a872c8a30d](https://github.com/cockroachdb/cockroach/commits/b3f25323eb94ac2759f2b51f440050a872c8a30d):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication3554963541
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564677399904257 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564677517901825 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564677636390913 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564677750063105 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564677854920705 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564677982748673 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564678093307905 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564678198034433 to reach replicated time of 1732132374.861516004,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022564678727860225 to reach replicated time of 1732132374.861516004,0
    jobs_verification.go:218: error scanning '&{<nil> 0x4006a4bd40}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication3554963541
--- FAIL: TestFourWayReplication (1.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17864882?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17864882?buildTab=artifacts#/) on master @ [c94b47e99eafa3cbae0c8684f11b7676abfd18fc](https://github.com/cockroachdb/cockroach/commits/c94b47e99eafa3cbae0c8684f11b7676abfd18fc):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication922156708
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588247481090049 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588247603150849 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588247732682753 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588247844913153 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588247946854401 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588248045617153 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588248141824001 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588248239210497 to reach replicated time of 1732139567.834095988,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022588248740069377 to reach replicated time of 1732139567.834095988,0
    jobs_verification.go:218: error scanning '&{<nil> 0x400bc95e60}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication922156708
--- FAIL: TestFourWayReplication (1.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17875119?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17875119?buildTab=artifacts#/) on master @ [89dae000e93ca462d99b251df47b1e162058b56c](https://github.com/cockroachdb/cockroach/commits/89dae000e93ca462d99b251df47b1e162058b56c):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication2609869905
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835900853190657 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835900992618497 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835901117136897 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835901345857537 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835901522280449 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835901665771521 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835901879156737 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835902016749569 to reach replicated time of 1732215145.721747829,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022835902158471169 to reach replicated time of 1732215145.721747829,0
    jobs_verification.go:218: error scanning '&{<nil> 0x401088fb00}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication2609869905
--- FAIL: TestFourWayReplication (2.21s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17881169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17881169?buildTab=artifacts#/) on master @ [56c767507b5d56780c6c69cc4432c2792e63da90](https://github.com/cockroachdb/cockroach/commits/56c767507b5d56780c6c69cc4432c2792e63da90):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication374575569
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942489810829313 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942489949306881 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490119438337 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490261618689 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490380861441 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490529398785 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490637336577 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490750320641 to reach replicated time of 1732247673.972964225,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022942490894598145 to reach replicated time of 1732247673.972964225,0
    jobs_verification.go:218: error scanning '&{<nil> 0x40084f79e0}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication374575569
--- FAIL: TestFourWayReplication (1.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/crosscluster/logical.TestFourWayReplication [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17881399?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17881399?buildTab=artifacts#/) on master @ [56c767507b5d56780c6c69cc4432c2792e63da90](https://github.com/cockroachdb/cockroach/commits/56c767507b5d56780c6c69cc4432c2792e63da90):


```
=== RUN   TestFourWayReplication
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication407539122
    test_log_scope.go:76: use -show-logs to present logs inline
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944694667542529 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944694772039681 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944694875389953 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944694978772993 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944695067705345 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944695167975425 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944695255433217 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944695342301185 to reach replicated time of 1732248346.848692591,0
    logical_replication_job_test.go:1438: waiting for logical replication job 1022944695824678913 to reach replicated time of 1732248346.848692591,0
    jobs_verification.go:218: error scanning '&{<nil> 0x4004510f30}': sql: no rows in result set
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a9cfd62e09137c7c70d57ded819e9790/logTestFourWayReplication407539122
--- FAIL: TestFourWayReplication (1.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135295 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3]
- #135124 ccl/crosscluster/logical: TestFourWayReplication failed [A-disaster-recovery C-test-failure O-robot P-2 T-disaster-recovery branch-release-24.3.0-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFourWayReplication.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go
+++ b/pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go
@@ -280,7 +280,9 @@ func createLogicalReplicationStreamPlanHook(
 		if cr, ok := options.GetDefaultFunction(); ok {
 			defaultConflictResolution = *cr
 		}
-		return p.ExecCfg().InternalDB.DescsTxn(ctx, func(ctx context.Context, txn descs.Txn) error {
+
+		var jobID jobspb.JobID
+		if err := p.ExecCfg().InternalDB.DescsTxn(ctx, func(ctx context.Context, txn descs.Txn) error {
 			dstTableDescs := make([]*tabledesc.Mutable, 0, len(srcTableDescs))
 			for _, pair := range repPairs {
 				dstTableDesc, err := txn.Descriptors().MutableByID(txn.KV()).Table(ctx, catid.DescID(pair.DstDescriptorID))
@@ -320,16 +322,19 @@ func createLogicalReplicationStreamPlanHook(
 				},
 				Progress: progress,
 			}
-
+			jobID = jr.JobID
 			if err := replicationutils.LockLDRTables(ctx, txn, dstTableDescs, jr.JobID); err != nil {
 				return err
 			}
 			if _, err := p.ExecCfg().JobRegistry.CreateAdoptableJobWithTxn(ctx, jr, jr.JobID, txn); err != nil {
 				return err
 			}
-			resultsCh <- tree.Datums{tree.NewDInt(tree.DInt(jr.JobID))}
-			return nil
-		})
+			return err
+		}); err != nil {
+			return err
+		}
+		resultsCh <- tree.Datums{tree.NewDInt(tree.DInt(jobID))}
+		return nil
 	}
 
 	return fn, streamCreationHeader, nil, false, nil

